textproc/lhs2tex/Portfile

```diff
@@ -1,22 +1,31 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           haskell 1.0
+PortGroup           haskell_stack 1.0
 
-haskell.setup       lhs2tex 1.18.1 ghc no
-name lhs2tex
-revision            3
-checksums           rmd160  047072143ddce8e83239a2e77c0f4fc6a2425132 \
-                    sha256  dccae22231a1db64cddc38b6f7ac98ac4d5677f03df2f548b2efe9a5e63e9648
+name                lhs2tex
+version             1.22
+checksums           rmd160  257e58b02249fd8403b14d314a519552afd2a5ab \
+                    sha256  7eea3b5b4bae019aa9acb6036e6c2606c62a0d9969a081d4b9b9f04baa3129bd \
+                    size    679988
 
 license             GPL-2+
 maintainers         {cal @neverpanic} openmaintainer
 platforms           darwin
-
+categories          textproc tex haskell
 description         Preprocessor for typesetting Haskell sources with LaTeX
 long_description    ${description}
 
-depends_lib-append  port:hs-mtl \
-                    port:hs-regex-compat \
-                    port:texlive-basic \
-                    port:texlive-latex-extra
+homepage            https://hackage.haskell.org/package/${name}
+master_sites        ${homepage}-${version}
+
+pre-build {
+    system -W ${workpath} \
+        "${prefix}/bin/stack init --stack-root [option haskell_stack.stack_root] \
+        --with-gcc ${configure.cc} \
+        --allow-different-user \
+        [haskell_stack.system_ghc_flags]"
+}
+
+livecheck.type      regex
```

Comment:
I think the livecheck is the default from the haskell_stack PortGroup, so it can probably be dropped?

Reply:
Yes. Thanks.